src/migrations/20230828103020-program-personnel-cleanup.js

```diff
@@ -0,0 +1,149 @@
+const {
+  prepMigration,
+  setAuditLoggingState,
+} = require('../lib/migration');
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      const sessionSig = __filename;
+      await prepMigration(queryInterface, transaction, sessionSig);
+
+      await queryInterface.sequelize.query(`
+
+        /* 1. Create a temp table of dedupe. */
+         DROP TABLE IF EXISTS "ProgramPersonnelToKeep";
+         CREATE TABLE "ProgramPersonnelToKeep" AS (
+            SELECT
+                min(id) id,
+                "programId",
+                "grantId",
+                role,
+                (ARRAY_AGG(prefix order by id desc))[1] prefix,
+                (ARRAY_AGG("firstName" order by id desc))[1] "firstName",
+                (ARRAY_AGG("lastName" order by id desc))[1] "lastName",
+                (ARRAY_AGG("suffix" order by id desc))[1] "suffix",
+                (ARRAY_AGG("title" order by id desc))[1] "title",
+                (ARRAY_AGG("email" order by id desc))[1] "email",
+                MAX("originalPersonnelId") "originalPersonnelId",
+                MIN("createdAt") "createdAt",
+                MAX("updatedAt") "updatedAt"
+                 FROM "ProgramPersonnel"
+                 GROUP BY
+                 "firstName",
+                 "lastName",
+                 "role",
+                 "grantId",
+                 "programId"
+         );`, { transaction });
+
+      /* 2. Disable audit trail. */
+      await setAuditLoggingState(queryInterface, transaction, false);
+
+      await queryInterface.sequelize.query(`
+         /* 3. Truncate botht the ZALProgramPersonnel and ProgramPersonnel tables. */
+         TRUNCATE TABLE "ZALProgramPersonnel";
+         TRUNCATE TABLE "ProgramPersonnel";
+       `, { transaction });
+
+      /* 4. Enable audit trail. */
+      await setAuditLoggingState(queryInterface, transaction, true);
+
+      await queryInterface.sequelize.query(`
+      /* 5. Rename originalProgramPersonnelId to mapsTo in the ProgramPersonnel table. */
+      ALTER TABLE "ProgramPersonnel" RENAME COLUMN "originalPersonnelId" TO "mapsTo";
```

Comment:
Please also change this from just an integer to a proper foreign key

Reply:
that still allows null

Reply:
done